server/controllers/request.js

```diff
@@ -0,0 +1,95 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+const {
+  updateRequest,
+  checkQueryForEmptyFields,
+  validateQueryForNewRequest,
+} = require("../utils/helperFunctions");
+
+// @route GET /request/load
+// @desc Get list of requests for logged-in user
+// @access Private
+exports.loadRequests = asyncHandler(async (req, res, next) => {
+  const uid = req.user.id;
+  let requests;
+
+  if (uid) {
+    requests = await Request.find({
+      userId: uid,
+    });
+  }
+
+  res.status(200).json(requests);
+});
+
+// @route POST /request/new
+// @desc Create a new request
+// @access Public
+exports.newRequest = asyncHandler(async (req, res, next) => {
+  const { validateQueryForNewRequest } = require("../utils/helperFunctions");
+
+  let data;
+  if (Object.keys(req.query).length > 0) {
+    data = req.query;
+  } else {
+    data = req.body;
+  }
```

Comment:
Could you elaborate on this conditional's purpose?

Reply:
Is it for allowing a new request to be created via query parameters? I am not a huge fan of this, we should enforce the request payload be sent down in the body.